src/apps/Permissions/Permissions.js

```diff
@@ -1,57 +1,27 @@
-import React from 'react'
+import React, { useEffect, useCallback, useRef, useState } from 'react'
 import PropTypes from 'prop-types'
-import styled from 'styled-components'
 import { AppType, AragonType } from '../../prop-types'
-import { IconPlus } from '@aragon/ui'
-import { addressesEqual, shortenAddress, isAddress } from '../../web3-utils'
-import Screen from './Screen'
+import { Button, GU, Header, IconPlus, Layout, useLayout } from '@aragon/ui'
+import { addressesEqual, isAddress } from '../../web3-utils'
+import { usePermissions } from '../../contexts/PermissionsContext'
+import LocalIdentityBadge from '../../components/IdentityBadge/LocalIdentityBadge'
+import useAppWidth from '../useAppWidth'
 import Home from './Home/Home'
 import AppPermissions from './AppPermissions'
-import EntityPermissions from './EntityPermissions'
-import NavigationItem from './NavigationItem'
 import AssignPermissionPanel from './AssignPermissionPanel'
 import ManageRolePanel from './ManageRolePanel'
-import { PermissionsConsumer } from '../../contexts/PermissionsContext'
-import AppLayout from '../../components/AppLayout/AppLayout'
 
-class Permissions extends React.Component {
-  static propTypes = {
-    apps: PropTypes.arrayOf(AppType).isRequired,
-    appsLoading: PropTypes.bool.isRequired,
-    onMessage: PropTypes.func.isRequired,
-    onParamsRequest: PropTypes.func.isRequired,
-    params: PropTypes.string,
-    permissionsLoading: PropTypes.bool.isRequired,
-    wrapper: AragonType,
+function getAppByProxyAddress(proxyAddress, apps) {
+  if (!proxyAddress) {
+    return null
   }
+  return apps.find(app => addressesEqual(app.proxyAddress, proxyAddress))
+}
 
-  state = {
-    // Only animate screens after the component is rendered once
-    animateScreens: false,
-    showAssignPermissionPanel: false,
-  }
-
-  componentDidMount() {
-    setTimeout(() => {
-      this.setState({ animateScreens: true })
-    }, 0)
-  }
-
-  componentDidUpdate(prevProps) {
-    const prevScreen = this.getLocation(prevProps.params).screen
-    const screen = this.getLocation(this.props.params).screen
-    if (prevScreen !== screen) {
-      this._scrollTopElement.scrollIntoView()
-    }
-  }
-
-  getLocation(params) {
-    const home = { screen: 'home' }
-
-    if (!params) {
-      return home
-    }
+function getLocation(params, apps) {
+  const home = { screen: 'home' }
 
+  if (params) {
     // Not using "/" as a separator because
```

Comment:
Hmmmmmm, maybe we could decode?

Reply:
`routing.js` is taking care of that :+1: https://github.com/aragon/aragon/blob/6db39231a265393c6e3db418a6d13035c32fe3b3/src/routing.js#L73-L79